android/src/main/java/com/gr4vy/embedreactnative/EmbedReactNativeModule.java

```diff
@@ -161,6 +166,39 @@ private static WritableArray convertCartItemsToJson(ReadableArray cartItemsArray
     return cartItemsWritableArray;
   }
 
+  private static JSONObject convertMapToJsonObject(ReadableMap map) {
```

Comment:
Since the object coming from React Native is a `ReadableMap`, which is not compatible with the `JSONObject` constructor, we need to manually iterate through it and convert it to a json object.